src/python/zquantum/core/interfaces/estimator_test.py

```diff
@@ -24,10 +58,45 @@ def test_get_estimated_expectation_values(self):
             # Given
             # When
             values = estimator.get_estimated_expectation_values(
-                self.backend, self.circuit, self.operator
+                self.backend,
+                self.circuit,
+                self.operator,
+                n_samples=self.n_samples,
+                epsilon=self.epsilon,
+                delta=self.delta,
             ).values
             value = values[0]
             # Then
             self.assertTrue(len(values) == 1)
             self.assertGreaterEqual(value, 0)
             self.assertLessEqual(value, 1)
+
+
+def parameter_is_ignored(
+    self, estimator, estimator_name, parameter_name, parameter_value
+):
+    with self.assertLogs("z-quantum-core", level="WARN") as context_manager:
+        # Given
+        expected_logs = [
+            "WARNING:z-quantum-core:{} = {}. {} does not use {}. The value was ignored.".format(
+                parameter_name, parameter_value, estimator_name, parameter_name
+            )
+        ]
+        # When
+        if parameter_name == "n_samples":
```

Comment:
Why do you override the self values of the parameters instead of just passing n_samples=n_samples in the estimator.get_estimated_expectation_values ?

Reply:
It depends on the parameter being ignored. When `parameter_name` is `"n_samples"`, then I want to do `n_samples=parameter_value`. Otherwise, I want to do `n_samples=self.n_samples'.

Reply:
Oh, makes sense.